### ISSUE TITLE
PTM-Mamba

### ISSUE THREAD

**Issue by JamesStrabala** [JamesStrabala/Capstone-Resorces#19]
### Title
PTM-Mamba: a PTM-aware protein language model with bidirectional gated Mamba blocks
### URL
https://doi.org/10.1038/s41592-025-02656-9
### Summary 
PTM-Mamba (PM) is a Mamba-based PLM that specializes in PTMs. (Post-transcriptional modifications, such as methylating or phosphorylating amino acids after they are chained) It was trained with 80K sequences with experimentally validated PTMs to give diversity of modifications and lengths. 
Mamba (or structured state-space models) were created to improve on the complexity of transformers by being one-directional. Like an RNN, it represents time and sequences as being discrete. With each new addition to the sequence, it uses replaces previously learned information with data from the new token, corresponding to the evolution of the system. 
Mamba is designed to be one-directional, but PM uses forward and backward mamba blocks to comprehend input. The output of the blocks is sent into a linear neural network to combine them. Input is embedded using ESM2's method and a framework designed for PTMs. Unaltered AAs use the corresponding token and PTMs become mask tokens. They are concatenated and filtered through a sigmoid function.
This setup converged faster on the training data than a PTM transformer. Stochastic neighbor embedding indicated that PM could learn nuanced distinctions between modified and unmodified AAs, and embedded PTMs could be grouped by their modification. When tested on PTM prediction, which does not need PTM tokens, PM had comparable performance to standard PLMs. It performed strongly when predicting PTM drugability, disease association, and PTM effect on protein-protein interaction. Zero-shot PTM discovery was also a strong area.
### Key Points 
* Mamba is an alternate approach to language models that is not transformer-based but performs well with proteins.
* First work I've found that centers around PTMs.
### Citation
@article{Peng_Wang_Chen_Schussheim_Vincoff_Chatterjee_2025, title={PTM-Mamba: A PTM-aware protein language model with bidirectional gated Mamba blocks}, DOI={10.1038/s41592-025-02656-9}, journal={Nature Methods}, author={Peng, Fred Zhangzhi and Wang, Chentong and Chen, Tong and Schussheim, Benjamin and Vincoff, Sophia and Chatterjee, Pranam}, year={2025}, month={Apr}}
